### PR TITLE
Add session close status in PubAddrs

### PR DESCRIPTION
**Pull request by billfort**
1. Use ncp.session.GetContext to follow ncp.session status;
2. Add session close status in PubAddrs to follow the listener's session status.

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -338,12 +338,16 @@ func (c *TunaSessionClient) getPubAddrsFromRemote(ctx context.Context, remoteAdd
 	if err != nil {
 		return nil, err
 	}
+	if pubAddrs.SessionClosed {
+		return pubAddrs, ErrClosed
```
Comment (yilunzhang) [inline client.go:342]: I think we should return `nil, ErrClosed`

File: client.go [hunk 2]
```diff
@@ -379,18 +383,21 @@ func (c *TunaSessionClient) listenNKN() {
 		if !c.shouldAcceptAddr(msg.Src) {
 			continue
 		}
+		log.Println("listenNKN msg.src is ", msg.Src)
```
Comment (yilunzhang) [inline client.go:386]: Is this for debug?

File: client.go [hunk 3]
```diff
@@ -379,18 +383,21 @@ func (c *TunaSessionClient) listenNKN() {
...
 			log.Printf("Decode request error: %v", err)
 			continue
 		}
-		switch strings.ToLower(req.Action) {
-		case "getpubaddr":
-			pubAddrs := c.getPubAddrs(false)
-			if len(pubAddrs.Addrs) == 0 {
-				log.Println("No entry available")
-				continue
+		switch req.Action {
```
Comment (yilunzhang) [inline client.go:393]: It's better to keep the case insensitive logic

File: client.go (hunk 1 above)
Comment (billfort) [reply client.go:342]: Yes, modified

File: client.go (hunk 2 above)
Comment (billfort) [reply client.go:386]: Yes, removed.

**Review by yilunzhang — APPROVED** [review]

**Pull request merged**